shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/domain/interactor/StudyPlanInteractor.kt

```diff
@@ -1,11 +1,27 @@
 package org.hyperskill.app.study_plan.domain.interactor
 
+import org.hyperskill.app.learning_activities.domain.model.LearningActivity
+import org.hyperskill.app.learning_activities.domain.model.LearningActivityType
+import org.hyperskill.app.learning_activities.domain.repository.LearningActivitiesRepository
 import org.hyperskill.app.study_plan.domain.model.StudyPlan
+import org.hyperskill.app.study_plan.domain.model.StudyPlanSection
 import org.hyperskill.app.study_plan.domain.repository.StudyPlanRepository
+import org.hyperskill.app.study_plan.domain.repository.StudyPlanSectionsRepository
 
 class StudyPlanInteractor(
-    private val studyPlanRepository: StudyPlanRepository
+    private val studyPlanRepository: StudyPlanRepository,
+    private val studyPlanSectionsRepository: StudyPlanSectionsRepository,
+    private val learningActivitiesRepository: LearningActivitiesRepository
 ) {
     suspend fun getCurrentStudyPlan(): Result<StudyPlan> =
         studyPlanRepository.getCurrentStudyPlan()
+
+    suspend fun getStudyPlanSections(sectionsIds: List<Long>): Result<List<StudyPlanSection>> =
+        studyPlanSectionsRepository.getStudyPlanSections(sectionsIds)
+
+    suspend fun getLearningActivities(
+        activitiesIds: List<Long>,
+        types: Set<LearningActivityType>
+    ): Result<List<LearningActivity>> =
+        learningActivitiesRepository.getLearningActivities(activitiesIds, types)
```

Comment:
In my opinion, it is redundant, because getLearningActivities does not contain any logic: it is just a request.
We can still use it via LearningActivitiesRepository.